### DISCUSSION TITLE
Extract all possible hydra keys from code.

### DISCUSSION THREAD

**Discussion by ysig** [facebookresearch/hydra#2494] [Q&A]
Is there a way that I can dry run a hydra based code and extract all the possible keys that can be given to the model.
Some of the defaults are not defined in the hydra file but in the code (and thus some keys are missing) so I was curious if I could lift them there in any way with small tweaks.

Thanks,

**Comment by Jasha10** [comment]
Hi @ysig,

Does the `--cfg=job` flag do what you're looking for?
```yaml
$ python my_app.py --cfg=job
db:
  driver: mysql
  user: omry
  password: secret
```
Reference: [Hydra's command line flags](https://hydra.cc/docs/1.2/advanced/hydra-command-line-flags/)

**Comment by ysig** [comment]
@Jasha10 Thank you for your answer. 

No this just collects the config into one file which I already have.

What I don't have is all the default values thare are of the form `config.get('lr', '0.001')` that are defined in the code and nowhere in the config.

Hydra could save all these things by overriding config with `'lr'=0.001` when get is called and `lr` is not in config.

Do you know if this supported yet or no?

Thanks!

**Reply by Jasha10** [reply]
No, that's not supported.
Once the `config` object is created (which usually has type `DictConfig`), Hydra doesn't play a role anymore. At that point it's a pure [OmegaConf](https://github.com/omry/omegaconf) object.

OmegaConf's `DictConfig` type tries to export an API that's similary to python's built-in `dict`.

**Reply by omry** [reply]
> What I don't have is all the default values thare are of the form config.get('lr', '0.001') that are defined in the code and nowhere in the config.

I recommend not doing that. Let Hydra fully compose the config you need with all the default values.
